### PR TITLE
[Fixes bug 1192815] Remove APIv1 code and UI

### PR DESCRIPTION
**Pull request by comzeradd**
* Update Settings UI to reflect the changes
* Remove unused fields from model and es
* Remove v1 references from documentation
* Remove tastypie from requirements

### CODE REVIEW

File: requirements/prod.txt [hunk 1]
```diff
@@ -29,9 +29,6 @@ django-sha2==0.4 \
     --hash=sha256:b06a86451a99d6c26bfa65082999dbf658e087e596e9292cce688b53eabaaada
 django-statsd-mozilla==0.3.8.2 \
     --hash=sha256:d305466c95eb580f82f8d1db14bcb90b42648671c274a8f4032ba24dbc7bb00e
-django-tastypie==0.12.2 \
```
Comment (johngian) [inline requirements/prod.txt]: You also need to remove the dependencies of tastypie.

File: mozillians/jinja2/phonebook/includes/profile_edit_developer.html [hunk 2]
```diff
@@ -1,56 +1,22 @@
 <div role="tabpanel" class="tab-pane fade" id="developer">
-  <form class="edit-profile" method="POST" action="{{ url('phonebook:profile_edit') }}?next=developer">
-    {% csrf_token %}
     <div class="panel panel-default">
       <div class="panel-heading">{{ _('API privacy') }}</div>
       <div class="panel-body">
-        {% if developer_form.non_field_errors() %}
-          <ul class="unstyled">
-            {% for error in developer_form.non_field_errors() %}
```
Comment (akatsoulas) [inline mozillians/jinja2/phonebook/includes/profile_edit_developer.html]: Is the developer_form only for the API v1?

File: mozillians/settings/local.py-devdist [hunk 3]
```diff
@@ -105,11 +105,6 @@ CSP_STYLE_SRC = base.CSP_STYLE_SRC + (
     'http://*.mozilla.org',
 )
 
-# Basket
-BASKET_URL = 'http://127.0.0.1'
-BASKET_API_KEY = 'basket_api_key'
-BASKET_MANAGERS = None  # or list of email addresses.
```
Comment (akatsoulas) [inline mozillians/settings/local.py-devdist]: Why are you removing this?

File: mozillians/users/es.py [hunk 4]
```diff
@@ -124,8 +122,7 @@ def extract_document(cls, obj_id, obj=None):
             obj = cls.get_model().objects.get(pk=obj_id)
         doc = {}
 
-        attrs = ('id', 'is_vouched', 'ircname',
-                 'allows_mozilla_sites', 'allows_community_sites')
+        attrs = ('id', 'is_vouched', 'ircname')
```
Comment (akatsoulas) [inline mozillians/users/es.py]: Let's add a `,` after `ircname`.

File: mozillians/settings/local.py-devdist (hunk 3 above)
Comment (comzeradd) [inline mozillians/settings/local.py-devdist]: Updating the settings I noticed these are duplicates. We already have the basket settings at line 71.

File: mozillians/jinja2/phonebook/includes/profile_edit_developer.html (hunk 2 above)
Comment (comzeradd) [inline mozillians/jinja2/phonebook/includes/profile_edit_developer.html]: Yes, APIv2 doesn't uses these fields at all. It uses privacy levels per field instead.

File: requirements/prod.txt (hunk 1 above)
Comment (comzeradd) [inline requirements/prod.txt]: Fixed. Remove two more dependencies that we don't require any more.

**Pull request merged**